miqa/core/tasks.py

```diff
@@ -56,90 +44,92 @@ def evaluate_data(image_ids, project_id):
-                            name=image_file,
-                            raw_path=image_dir / image_file,
-                            scan=scan,
-                        )
-                    )
-
-    Scan.objects.bulk_create(scans)
-    Image.objects.bulk_create(images)
-    ScanNote.objects.bulk_create(notes)
-    Annotation.objects.bulk_create(annotations)
 
-    evaluate_data.delay([image.id for image in images], project.id)
+        # TODO: put this back for support of multiple projects in one import
+        # all_projects.append({
+        #     project_name: Project(name=project_name, creator=user)
+        #     for project_name in df['project_id'].unique()
+        # })
+        # for project_name, project_object in all_projects[0].items():
+        for project_name, project_object in [(df['project_id'].mode()[0], project)]:
+
+            # delete old imports of these projects
+            Experiment.objects.filter(
+                project=project_object
+            ).delete()  # cascades to scans -> images, scan_notes
+
+            these_experiments = {
+                experiment_name: Experiment(name=experiment_name, project=project_object)
+                for experiment_name in df[df['project_id'] == project_name][
+                    'experiment_id'
+                ].unique()
+            }
+            all_experiments.append(these_experiments)
+            for experiment_name, experiment_object in these_experiments.items():
+                these_scans = {
+                    scan_name: Scan(
+                        name=scan_name, scan_type=scan_type, experiment=experiment_object
+                    )
+                    for scan_name, scan_type in df[df['experiment_id'] == experiment_name]
+                    .groupby(['scan_id', 'scan_type'])
+                    .groups
+                }
+                all_scans.append(these_scans)
+                for scan_name, scan_object in these_scans.items():
+                    these_frames = {}
+                    for row in df[df['scan_id'] == scan_name].iterrows():
+                        raw_path = row[1]['file_location']
+                        if raw_path[0] != '/':
+                            # not an absolute file path; refer to project import csv location
+                            # TODO: add support for interpreting URIs not on host machine
+                            raw_path = str(Path(project.import_path).parent.parent) + '/' + raw_path
+                        these_frames[row[1]['frame_number']] = Image(
+                            frame_number=int(row[1]['frame_number']),
+                            scan=scan_object,
+                            raw_path=raw_path,
+                        )
+                    all_frames.append(these_frames)
+
+        for model, list_all in [
+            (Project, all_projects),
+            (Experiment, all_experiments),
+            (Scan, all_scans),
+            (Image, all_frames),
```

Comment:
```
            (Experiment, new_experiments),
            (Scan, new_scans),
            (Image, new_frames),
```
Since these are apparently only used to create newly imported models, I think calling them `new_` rather than `all_` is a little more legible.